src/libos/src/process/do_vfork.rs

```diff
@@ -201,3 +228,62 @@ pub fn handle_force_stop() {
         }
     }
 }
+
+// Wait4 unwaited child which are created with vfork and directly exit without calling execve.
+pub fn wait4_exit_child_created_with_vfork(
+    parent_pid: pid_t,
+    child_filter: &ProcessFilter,
+) -> Result<(pid_t, i32)> {
+    let mut children_status = EXIT_CHILDREN_STATUS.lock().unwrap();
+    if let Some(children) = children_status.get_mut(&parent_pid) {
+        let unwaited_child_idx = children.iter().position(|child| match child_filter {
+            ProcessFilter::WithAnyPid => true,
+            ProcessFilter::WithPid(pid) => pid == child.pid(),
+            ProcessFilter::WithPgid(pgid) => todo!(), // This case should be rare.
+        });
+
+        if let Some(child_idx) = unwaited_child_idx {
+            let child = children.remove(child_idx);
+            if children.is_empty() {
+                children_status.remove(&parent_pid);
+            }
+            return Ok((*child.pid(), child.status().as_u32() as i32));
+        }
+    }
+
+    return_errno!(ECHILD, "no exit child found");
+}
+
+// Reap all unwaited child which are created with vfork and directly exit without calling execve.
+pub fn reap_zombie_child_created_with_vfork(parent_pid: pid_t) -> Vec<pid_t> {
```

Comment:
reap_zombie_child_created_with_vfork(parent_pid: pid_t) -> Option<Vec<pid_t>>

Reply:
Done. Thanks for your review.